etl/steps/data/garden/worldbank_wdi/2024-05-20/wdi.py

```diff
@@ -77,6 +83,73 @@ def run(dest_dir: str) -> None:
     log.info("wdi.end")
 
 
+def add_regions_to_remittance_data(tb: Table, ds_regions: Dataset, ds_income_groups: Dataset) -> Table:
+    """
+    Add regions to remittance data.
+    """
+
+    tb = tb.reset_index()
+
+    # create a copy so other indicators are not affected
+    regions_tb = tb.copy()
+
+    # create new columns for total remittances (only for countries where remittance cost is available)
+    regions_tb["total_received_remittances"] = regions_tb["bx_trf_pwkr_cd_dt"].where(
+        regions_tb["si_rmt_cost_ib_zs"].notna()
+    )
+    regions_tb["total_sent_remittances"] = regions_tb["bm_trf_pwkr_cd_dt"].where(
+        regions_tb["si_rmt_cost_ob_zs"].notna()
+    )
+
+    # calculate total cost of remittance for each country
+    regions_tb["total_cost_of_receiving_remittances"] = (
+        regions_tb["si_rmt_cost_ib_zs"] * regions_tb["total_received_remittances"]
+    )
+    regions_tb["total_cost_of_sending_remittances"] = (
+        regions_tb["si_rmt_cost_ob_zs"] * regions_tb["total_sent_remittances"]
+    )
+
+    agg = {
+        "total_cost_of_receiving_remittances": "sum",
+        "total_cost_of_sending_remittances": "sum",
+        "total_received_remittances": "sum",
+        "total_sent_remittances": "sum",
+        "bx_trf_pwkr_cd_dt": "sum",
+        "bm_trf_pwkr_cd_dt": "sum",
+    }
+
+    # add regions to table
+    regions_tb = geo.add_regions_to_table(
+        regions_tb,
+        ds_regions=ds_regions,
+        ds_income_groups=ds_income_groups,
+        aggregations=agg,
+        min_num_values_per_year=1,
```

Comment:
For me this is a very permissive restriction, that can maybe not useful fore regional analysis. I am more inclined to using frac_allowed_nans_per_year

Reply:
I've added a filter for the regional average calculation afterwards - only include the average if more than 75% of transaction volume has an average cost associated.